src/platform/Darwin/KeyValueStoreManagerImpl.mm

```diff
@@ -240,22 +251,36 @@ - (instancetype)initWithContext:(nonnull NSManagedObjectContext *)context
         CHIP_ERROR KeyValueStoreManagerImpl::_Put(const char * key, const void * value, size_t value_size)
         {
             ReturnErrorCodeIf(key == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+            ReturnErrorCodeIf(gContext == nullptr, CHIP_ERROR_WELL_UNINITIALIZED);
 
             NSData * data = [[NSData alloc] initWithBytes:value length:value_size];
 
             KeyValueItem * item = FindItemForKey([[NSString alloc] initWithUTF8String:key], nil);
-            if (!item) {
-                item = [[KeyValueItem alloc] initWithContext:gContext key:[[NSString alloc] initWithUTF8String:key] value:data];
-                [gContext insertObject:item];
-            } else {
-                item.value = data;
+            if (item) {
+                // For some reason just updating the state of "item" does not seem
+                // to work right, with complaints about using a managed object
+                // associated with a no-longer-existing context.
+                //
+                // Need to sort out what's going on there, but in the meantime
+                // just remove this item and create a new one in its place.
+                [gContext deleteObject:item];
```

Comment:
Just curious, does it work with the fix in https://github.com/project-chip/connectedhomeip/pull/14390 ?